cmd/util/common.go

```diff
@@ -1,6 +1,38 @@
 package util
 
+import (
+	stderrors "errors"
+
+	"github.com/argoproj/argo-cd/v3/util/errors"
+)
+
 var (
 	LogFormat string
 	LogLevel  string
 )
+
+func ValidateBearerTokenForHTTPSRepoOnly(bearerToken string, isHTTPS bool) {
```

Comment:
WDYT about returning errors in validate functions, so you will be able to build better unit testing around it? and you will not need to check output

Reply:
My incentive for that was to avoid the code duplication in all the calling functions that would check whether err is nil and invoking `errors.CheckError(err)`. I preferred to have the testing a bit more complicated over duplicating those 2 lines of code in the actual logic. Do you see any limitation to unit testing in the current implementation?